www/js/diary.js

```diff
@@ -352,6 +353,44 @@ angular.module('emission.main.diary',['ui-leaflet', 'nvd3ChartDirectives',
     };
     */
 
+    /* For UI control */
+    $scope.groups = [];
+    for (var i=0; i<10; i++) {
+      $scope.groups[i] = {
+        name: i,
+        items: ["good1", "good2", "good3"]
+      };
+      for (var j=0; j<3; j++) {
+        $scope.groups[i].items.push(i + '-' + j);
+      }
+    }
+    $scope.toggleGroup = function(group) {
+      if ($scope.isGroupShown(group)) {
+        $scope.shownGroup = null;
+      } else {
+        $scope.shownGroup = group;
+      }
+    };
+    $scope.isGroupShown = function(group) {
+      return $scope.shownGroup === group;
+    };
+    $scope.allModes = function(trip) {
+      var rtn = [];
+      var icons = {"BICYCLING":"ion-android-bicycle",
+                    "WALKING":" ion-android-walk",
+                    "RUNNING":" ion-android-walk",
+                    "IN_VEHICLE":"ion-disc",}
+      for (var i=0; i<trip.sections.length; i++) {
+        if (rtn.indexOf(trip.sections[i].properties.sensed_mode.split('.')[1]) == -1) {
```

Comment:
Can reuse $scope.getHumanReadable(sensed_mode) here. Will make the change while merging.